src/Material.cc

```diff
@@ -358,3 +366,64 @@ void Material::SetFilePath(const std::string &_filePath)
 {
   this->dataPtr->filePath = _filePath;
 }
+
+//////////////////////////////////////////////////
+ignition::math::Color Material::Implementation::GetColor(
+    const std::string &_colorType, Errors &_errors)
+{
+  ignition::math::Color color(0, 0, 0, 1);
+
+  if (!this->sdf->HasElement(_colorType))
+    return color;
+
+  std::string colorStr = this->sdf->GetElement(_colorType)
+                                  ->GetValue()
+                                  ->GetAsString();
+  if (colorStr.empty())
+    return color;
+
+  std::stringstream ss(colorStr);
+  std::string token;
+  std::vector<float> colors;
+  float c;  // r,g,b,a values
+  bool isValidColor = true;
+  while (ss >> token)
+  {
+    try
+    {
+      c = std::stof(token);
+      colors.push_back(c);
+    }
+    catch(const std::exception &e)
+    {
+      std::cerr << "Error converting color value: can not convert ["
+                << token << "] to a float" << std::endl;
+      isValidColor = false;
+      break;
+    }
+
+    if (c < 0.0f || c > 1.0f)
+    {
+      isValidColor = false;
+      break;
+    }
+  }
+
+  size_t colorSize = colors.size();
+  if (colorSize == 3)
+    colors.push_back(1.0f);
```

Comment:
It looks like the default alpha value for a `rgb` input is `1.0`. Should we add a note about this behavior to the method's documentation?

Reply:
f4633ca1011bd63afe05cb51928a5736a52fa09c